### PR TITLE
fix: windows gitignore

### LINKED ISSUE #158: `filter.respect_gitignore = true` seems not working on WindowsOS
also when I set `filter.show_hidden = false` my `.gitignore` doesn't render, is this intended?
Edit: I did notice that just not works with directories

### PR DESCRIPTION
**Pull request by lopi-py**
fixes #158

### CODE REVIEW

File: lua/neo-tree/utils.lua [hunk 1]
```diff
@@ -347,13 +347,6 @@ M.resolve_config_option = function(state, config_option, default_value)
   end
 end
 
----The file system path separator for the current platform.
```
Comment (cseickel) [inline lua/neo-tree/utils.lua:350]: This part should not be removed, it is used in many places.

File: lua/neo-tree/git/ignored.lua [hunk 2]
```diff
@@ -68,7 +68,7 @@ end
 M.is_ignored = function(ignored, path, _type)
   path = _type == "directory" and (path .. sep) or path
   for _, v in ipairs(ignored) do
-    if v:sub(-1) == utils.path_separator then
+    if v:sub(-1) == utils.path_separator or (utils.is_windows and _type == "directory") then
```
Comment (cseickel) [inline lua/neo-tree/git/ignored.lua:71]: How does this work if `utils.is_windows` was removed?

File: lua/neo-tree/utils.lua (hunk 1 above)
Comment (lopi-py) [reply lua/neo-tree/utils.lua:350]: it's duplicated, see
https://github.com/nvim-neo-tree/neo-tree.nvim/blob/main/lua/neo-tree/utils.lua#L350
and
https://github.com/nvim-neo-tree/neo-tree.nvim/blob/main/lua/neo-tree/utils.lua#L384

File: lua/neo-tree/git/ignored.lua (hunk 2 above)
Comment (lopi-py) [reply lua/neo-tree/git/ignored.lua:71]: Yes, but it's just tested on windows, don't know if will work on linux or macOS

File: lua/neo-tree/utils.lua (hunk 1 above)
Comment (cseickel) [reply lua/neo-tree/utils.lua:350]: OH! Good catch!

**Review by cseickel — APPROVED** [review]
Thanks for contributing!

**Pull request merged**